lib/rspec/matchers/built_in/compound.rb

```diff
@@ -155,6 +155,13 @@ def initialize(actual, matcher_1, matcher_2)
 
           def matcher_matches?(matcher)
             @match_results.fetch(matcher)
+          rescue IndexError
+            raise ArgumentError,
+              """
+                A matcher has no match results, this can occur when an unexpected
```

Comment:
"A matcher has no match results" is kinda vague.  Would be good to use `matcher.description` in here somewhere to be more specific.  Maybe: 

```
"Your '#{matcher.description}' matcher was unable to produce a match result. " \
"Usually this happens when one of the matchers in your compound expression " \
"experiences a call stack jump (e.g. via a `throw` or `raise`) without defining " \
"`expects_call_stack_jump?` to return `true`.
```

Also, notice my use of `\`; that concats the multi-line string at parse time.  I think it's better to have a single long string that wraps in the user's terminal based on whatever their terminal width is rather than hard-wrapping it for them based on the width you decided on here.